File: PS/Sliding Window/1004. Max Consecutive Ones III/Solution.dart
import 'dart:math';

class Solution {
  int longestOnes(List<int> nums, int k) {
      var left = 0, right = 0;
      var answer = 0;
      var numOfZero = 0;

      while (right < nums.length) {
        if (nums[right] == 0) numOfZero += 1;

        while (numOfZero > k) {
          if (nums[left] == 0) numOfZero -= 1;
          left += 1;
        }

        answer = max(answer, right - left + 1);
        right += 1;
      }

      return answer;
  }
}

File: PS/Sliding Window/1004. Max Consecutive Ones III/Solution.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        left, right = 0, 0
        num_zero = 0
        answer = 0

        while right < len(nums):
            num_zero += nums[right] == 0

            while num_zero > k:
                num_zero -= nums[left] == 0
                left += 1
            
            answer = max(answer, right - left + 1)
            right += 1

        return answer